feat(curriculum): add Blockquotes and Quote Workshop to FSD cert

l3onhard opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes [#982](https://github.com/freeCodeCamp/CurriculumExpansion/issues/982)

<!-- Feel free to add any additional description of changes below this line -->

l3onhard committed:
feat(curriculum): add Blockquotes and Quote Workshop to FSD cert

ilenia-magoni commented on client/i18n/locales/english/intro.json:
```diff
@@ -5013,6 +4980,10 @@
         "title": "Certified Full Stack Developer Exam",
         "intro": ["Pass this exam to become a Certified Full Stack Developer."]
+      },
+      "workshop-quincys-job-tips": {
```
can you move this to right above "lecture-working-with-specialized-semantic-elements"? same order as in full-stack-developer.json

l3onhard committed:
fix: move workshop-quincys-job-tips to correct place

l3onhard committed:
fix(curriculum): change demoType of workshop-quincys-job-tips and remove comment

l3onhard committed:
feat(curriculum): copied step 1 from workshop-blog-page

l3onhard committed:
feat(curriculum): adapt desciption of step 1

l3onhard committed:
feat(curriculum): simplify regex for testing the order of elements

l3onhard committed:
feat(curriculum): add step 2 to workshop-quincys-job-tips

l3onhard committed:
feat(curriculum): copied step 2 from workshop-blog-page

l3onhard committed:
feat(curriculum): simplify regex for testing the order of elements in step 2

l3onhard committed:
feat(curriculum): remove unnecessary brackets in regex around \s

l3onhard committed:
fix(curriculum): adapt step 1 to step 1 of workshop-build-a-video-display-using-iframe

l3onhard committed:
fix(curriculum): adapt step 2 to add p element with text inside

l3onhard committed:
feat(curriculum): create step 3 with script

l3onhard committed:
feat(curriculum): adapt seed and add description to step 3

l3onhard committed:
feat(curriculum): create step 4 with script

l3onhard committed:
fix(curriculum): expanded the introduction of the q element in step 3

l3onhard committed:
feat(curriculum): adapt seed and add description to step 4

l3onhard committed:
fix(curriculum): make instruction clearer in step 3

l3onhard committed:
feat(curriculum): create step 5 with script

l3onhard committed:
feat(curriculum): adapt seed and add description to step 5

l3onhard committed:
feat(curriculum): create step 6 with script

l3onhard committed:
feat(curriculum): adapt seed and add description to step 6

l3onhard committed:
feat(curriculum): create step 7 with script

l3onhard committed:
feat(curriculum): adapt seed and add description to step 7

l3onhard committed:
feat(curriculum): create step 8 with script

l3onhard committed:
feat(curriculum): adapt seed and add description to step 8

l3onhard committed:
fix(curriculum): copy exmaple quote from step 8 to step 7

l3onhard committed:
feat(curriculum): create step 9 with script

l3onhard committed:
feat(curriculum): adapt seed and add description to step 9

l3onhard committed:
feat(curriculum): create step 10 with script

l3onhard committed:
fix(curriculum): remove unnecessary dot after book title in step 9

l3onhard committed:
feat(curriculum): adapt seed and add description to step 10

l3onhard committed:
feat(curriculum): create step 11 with script

l3onhard committed:
feat(curriculum): adapt seed and add description to step 11

l3onhard committed:
feat(curriculum): create step 12 with script

l3onhard committed:
fix(curriculum): add instruction for cite attribute in step 11

l3onhard committed:
feat(curriculum): adapt seed and add description to step 12

l3onhard committed:
fix(curriculum): make test more permissive in step 2

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

l3onhard committed:
fix(curriculum): adapt test to use css selector in step 2

l3onhard committed:
Merge branch 'feat/add-blockquotes-and-quote-workshop-to-fsd-cert' of github.com:l3onhard/freeCodeCamp into feat/add-blockquotes-and-quote-workshop-to-fsd-cert
fix(curriculum): merge an accepted change suggestion on pr 62568

l3onhard committed:
Merge branch 'freeCodeCamp:main' into feat/add-blockquotes-and-quote-workshop-to-fsd-cert

l3onhard committed:
fix(curriculum): small corrections in description of step 12

l3onhard committed:
feat(curriculum): create step 13 with script

l3onhard committed:
feat(curriculum): adapt seed and add description to step 13

l3onhard committed:
feat(curriculum): add solution to step 13

l3onhard committed:
fix(curriculum): small corrections in descriptions of steps 6, 10 and 12

l3onhard committed:
fix(curriculum): remove wrong cite attribute in step 10

l3onhard committed:
feat(curriculum): add tests to step 3

l3onhard committed:
feat(curriculum): add tests to step 4

l3onhard committed:
feat(curriculum): add tests to steps 5-13

l3onhard committed:
fix(curriculum): make test more permissive in step 1

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

l3onhard committed:
fix(curriculum): add missing intros to intro.json and index.md

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68e96ce52087043b0e999296.md:
````diff
@@ -0,0 +1,54 @@
+
+```js
+const thirdSectionEl = document.querySelector('main > section:nth-of-type(3)');
+assert.exists(thirdSectionEl);
````
using `assert.lengthOf` would be better here because it matches the approach in the other workshops

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68eab6a2b37d2f13602d0c3f.md:
````diff
@@ -0,0 +1,110 @@
+
+```js
+const forthPEl = document.querySelector('main > section:nth-of-type(2) > blockquote > p:nth-of-type(4)');
+assert.exists(forthPEl);
````
you should use `assert.lengthOf` to match the other workshop tests

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68ebe9cd7523d340c0b343d3.md:
````diff
@@ -0,0 +1,192 @@
+```js
+const thirdPEl = document.querySelector('main > section:nth-of-type(3) > blockquote > p:nth-of-type(3)');
+assert.exists(thirdPEl);
+```
````
you should use `assert.lengthOf` to match the other workshop tests

l3onhard replied:
[fix(curriculum): change element counting tests to assert.lengthOf](https://github.com/freeCodeCamp/freeCodeCamp/pull/62568/commits/a52d30efce070ce4ffd784fc78ca6d70e0da60d2)

jdwilkin4 commented on curriculum/structure/blocks/workshop-quincys-job-tips.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "Build a Quincy's Job Tips Page",
+  "isUpcomingChange": true,
```
This should be changed to false. we only set it to true for content that is not ready to be live on learn. But in this case, once this is merged in and deployed, campers can start working on it

```suggestion
  "isUpcomingChange": false,
```

jdwilkin4 reviewed (commented):
Went through it locally and left some comments 👍🏾

l3onhard committed:
fix(curriculum): replace all occurrences of the workshop title

l3onhard commented on client/i18n/locales/english/intro.json:
```diff
@@ -2214,6 +2214,12 @@
         ]
       },
+      "workshop-quincys-job-tips": {
+        "title": "Build a Quincy's Job Tips Page",
```
[fix(curriculum): replace all occurrences of the workshop title](https://github.com/freeCodeCamp/freeCodeCamp/pull/62568/commits/72b4cfc460b8c5d165f3bcad426cac5d92c88efa)

l3onhard committed:
fix(curriculum): simplify and adapt wording in descriptions and hints

Co-authored-by: Jessica Wilkins  <67210629+jdwilkin4@users.noreply.github.com>

l3onhard committed:
fix(curriculum): set the workshop's isUpcomingChange to false

Co-authored-by: Jessica Wilkins  <67210629+jdwilkin4@users.noreply.github.com>

l3onhard committed:
fix(curriculum): shorten introductory text for q element

Co-authored-by: Jessica Wilkins  <67210629+jdwilkin4@users.noreply.github.com>

l3onhard commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68e90335bd53a901c41dfc13.md:
````diff
@@ -0,0 +1,70 @@
+assert.equal(document.querySelector('h1 + p > q')?.textContent.trim(), 'You can become a developer.');
+```
+
+Ignoring the `q` element you just added, the text of the `p` element should still be `Learning to code is hard, but as Quincy Larson says, You can become a developer.`.
````
Should the "y" in "[...] says, you can become [...]" be uppercase? It is uppercase in the HTML.

l3onhard committed:
fix(curriculum): change element counting tests to assert.lengthOf

ilenia-magoni commented:
please don't push for each commit, even if they are stopped soon, the CI still goes in the queue and consume resources while it goes

if you are accepting suggestions, add them to a batch first from the Files changed tab

jdwilkin4 commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68ebe9cd7523d340c0b343d3.md:
```diff
@@ -0,0 +1,192 @@
+
+# --hints--
+
+Your should have an `h2` element nested inside the thrid section.
```
```suggestion
Your should have an `h2` element nested inside the third section.
```

jdwilkin4 committed:
Update curriculum/challenges/english/blocks/workshop-quincys-job-tips/68ebe9cd7523d340c0b343d3.md

jdwilkin4 reviewed (approved):
Went through it again and found a small typo. So I committed that fix to expedite the review process. 

LGTM 👍🏾

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68e964b84f954c322376e592.md:
```diff
@@ -0,0 +1,65 @@
+
+# --description--
+
+If the source of a quote is a website, you can use cite it with the `cite` element. The value of this attribute should be a valid URL. While this attribute doesn't change the presentation of the block quote, it's very helpful for giving screen readers and search engines more information about the quote.
```
```suggestion
If the source of a quote is a website, you can cite it with the `cite` attribute. The value of this attribute should be a valid URL. While this attribute doesn't change the presentation of the block quote, it's very helpful for giving screen readers and search engines more information about the quote.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68ea7e7b916e8270483689b6.md:
```diff
@@ -0,0 +1,74 @@
+
+# --hints--
+
+Inside the first section, you should have a `blockquote` element below the `h2` element.
```
```suggestion
Inside the first `section` element, you should have a `blockquote` element below the `h2` element.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68ea7e7b916e8270483689b6.md:
````diff
@@ -0,0 +1,74 @@
+assert.exists(document.querySelector('section:first-of-type > h2 + blockquote'));
+```
+
+Your blockquote should have the text `Can you imagine what it would be like to be a successful developer? To have built software systems that people rely upon?`.
````
```suggestion
Your `blockquote` element should have the text `Can you imagine what it would be like to be a successful developer? To have built software systems that people rely upon?`.
```

l3onhard committed:
fix(curriculum): apply suggestions from code review

- fix typos
- standardise wording
- make the q element textContent test stricter

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

l3onhard commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68e90335bd53a901c41dfc13.md:
````diff
@@ -0,0 +1,70 @@
+Your `q` element should have the text `You can become a developer.`.
+
+```js
+assert.equal(document.querySelector('h1 + p > q')?.textContent.trim(), 'You can become a developer.');
````
I prefer the version without white space before and after the text. I will push a commit with the necessary changes.

l3onhard committed:
fix(curriculum): ensure q content has no leading/trailing whitespace

a2937 committed:
Merge branch 'main' into feat/add-blockquotes-and-quote-workshop-to-fsd-cert

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68e964b84f954c322376e592.md:
````diff
@@ -0,0 +1,65 @@
+assert.exists(qEl?.getAttribute('cite'));
+```
+
+Your `cite` attribute of the `q` element should have the value `https://www.freecodecamp.org/news/learn-to-code-book/`.
````
```suggestion
The `cite` attribute of your `q` element should have the value `https://www.freecodecamp.org/news/learn-to-code-book/`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68eab45aceb7c00fd8de4fed.md:
````diff
@@ -0,0 +1,97 @@
+assert.exists(blockquoteEl?.getAttribute('cite'));
+```
+
+Your `cite` attribute of the `blockquote` element should have the value `https://www.freecodecamp.org/news/learn-to-code-book/`.
````
```suggestion
The `cite` attribute of your `blockquote` element should have the value `https://www.freecodecamp.org/news/learn-to-code-book/`.
```

ilenia-magoni commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68ebe9cd7523d340c0b343d3.md:
````diff
@@ -0,0 +1,192 @@
+assert.equal(blockquoteEl?.getAttribute('cite'), 'https://www.freecodecamp.org/news/learn-to-code-book/');
+```
+
+You should have three `p` elements nested inside the `blockquote` element inside the third section.
````
```suggestion
You should have three `p` elements nested inside the `blockquote` element in the third section.
```

majestic-owl448 reviewed (commented):
found some more small corrections needed.

also I think the last step may need a "Congratulations! You have finished the workshop" or similar sentence

l3onhard commented on curriculum/challenges/english/blocks/workshop-quincys-job-tips/68e90335bd53a901c41dfc13.md:
````diff
@@ -0,0 +1,76 @@
+You should only add `q` tags around the appropriate part of the sentence without changing the original text. The `p` element should still read: `Learning to code is hard, but as Quincy Larson says, you can become a developer.`
+
+```js
+const pElCleanedText = document.querySelector('h1 + p')?.innerText.trim().replace(/ {2,}/g, ' ');
````
As I understand it `replaceAll(pattern, replacement)` does the same thing as `replace(pattern, replacement)`, if the pattern is a global regex: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/Symbol.replace#description
I tested it here: https://codepen.io/liontooth/pen/zxrjqgp

l3onhard committed:
fix(curriclum): apply suggestions from ilenia-magoni's code review

- correct grammatical errors
- improve wording

Co-authored-by: Ilenia <26656284+ilenia-magoni@users.noreply.github.com>

l3onhard committed:
fix(curriculum): add sentence to mark the end of the workshop

l3onhard committed:
fix(curriculum): update field blockType to blockLabel

Co-authored-by: Huyen Nguyen <25715018+huyenltnguyen@users.noreply.github.com>

majestic-owl448 committed:
Apply suggestions from code review

moT01 reviewed (approved):
LGTM @l3onhard 🎉 Thanks for contributing to freeCodeCamp 🎉
